pkg/controller/subscription/subscription_controller.go

```diff
@@ -253,23 +245,21 @@ func (r *ReconcileSubscription) HandleUpgrades(ctx context.Context, rhmiSubscrip
 
 	isServiceAffecting := rhmiConfigs.IsUpgradeServiceAffecting(latestRHMICSV)
 
-	// checks if there is changes in the rhmiconfig cr and recalculate the status
-	if reschedule, ok := config.Annotations[v1alpha1.RecalculateScheduleAnnotation]; ok && reschedule == "true" {
-		err = rhmiConfigs.UpdateStatus(ctx, r.client, config, latestRHMIInstallPlan, rhmiSubscription.Status.CurrentCSV)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-
-		config.Annotations[v1alpha1.RecalculateScheduleAnnotation] = ""
-		err = r.client.Update(ctx, config)
-		if err != nil {
-			return reconcile.Result{}, err
+	if isServiceAffecting {
+		newUpgradeAvailable := &integreatlyv1alpha1.UpgradeAvailable{
+			TargetVersion: rhmiSubscription.Status.CurrentCSV,
+			AvailableAt:   latestRHMIInstallPlan.CreationTimestamp,
 		}
 
-	} else if isServiceAffecting && rhmiSubscription.Status.CurrentCSV != config.Status.TargetVersion {
-		err = rhmiConfigs.UpdateStatus(ctx, r.client, config, latestRHMIInstallPlan, rhmiSubscription.Status.CurrentCSV)
-		if err != nil {
-			return reconcile.Result{}, err
+		if !reflect.DeepEqual(newUpgradeAvailable, config.Status.UpgradeAvailable) && !latestRHMIInstallPlan.Spec.Approved {
+			config.Status.UpgradeAvailable = newUpgradeAvailable
+			if err := r.client.Status().Update(context.TODO(), config); err != nil {
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{
+				Requeue:      true,
+				RequeueAfter: 10 * time.Second,
+			}, nil
```

Comment:
After verifying the patch upgrade recalculation I realized this could be simplified and another condition would be needed in order to avoid recalculation when a patch release is approved:

```go
if isServiceAffecting && !latestRHMIInstallPlan.Spec.Approved && config.Status.UpgradeAvailable == nil {
		config.Status.UpgradeAvailable = &integreatlyv1alpha1.UpgradeAvailable{
			TargetVersion: rhmiSubscription.Status.CurrentCSV,
			AvailableAt:   latestRHMIInstallPlan.CreationTimestamp,
		}
		if err := r.client.Status().Update(context.TODO(), config); err != nil {
			return reconcile.Result{}, err
		}
		return reconcile.Result{
			Requeue:      true,
			RequeueAfter: 10 * time.Second,
		}, nil
	}
```

Mainly, if the UpgradeAvailable field is already set, there's no need to update it. When the upgrade is approved this field will be set to `nil`, so subsequent upgrades will still be scheduled